### PR TITLE
Move EthClient and ethereum specific logic into separate eth/ pkg

### PR DESCRIPTION
**Pull request by dimroc**
https://www.pivotaltracker.com/story/show/170019505

Allows easier consumption of ethereum functionality without the need to pull in Chainlink specific logic. Helpful for FM. Lightens the store and models package.

Types moved and renamed, destination pkg `eth` unless stated otherwise:
1. models.FunctionSelector -> eth.FunctionSelector
2. store.EthClient -> eth.Client
3. store.EthCallerSubscriber -> eth.CallerSubscriberClient
4. models.EthSubscription, EthSubscriber, LogSubscriber -> eth.Subscription, Subscriber, LogSubscriber
4. models.Big -> utils.Big
5. models.TxReceipt -> eth.TxReceipt
6. models.BlockHeader -> eth.BlockHeader
7. core/store/assets -> core/assets

Can now easily reuse `ethClient := &eth.CallerSubscriberClient{CallerSubscriber: ethrpc}` without pulling in CL specific packages.

### CODE REVIEW

**Comment by felder-cl** [comment]
Looks good, but unfortunately I disagree about the renaming.  I believe the time to remove the prepended 'Eth's would be now.

File: core/internal/mocks/eth_client.go [hunk 1]
```diff
@@ -2,27 +2,32 @@
 
 package mocks
 
-import assets "chainlink/core/store/assets"
-import big "math/big"
-import common "github.com/ethereum/go-ethereum/common"
-import ethereum "github.com/ethereum/go-ethereum"
-import mock "github.com/stretchr/testify/mock"
-import models "chainlink/core/store/models"
```
Comment (felder-cl) [inline core/internal/mocks/eth_client.go]: :+1:

File: core/internal/mocks/eth_client.go [hunk 2]
```diff
@@ -2,27 +2,32 @@
...
+)
+
+// Client is an autogenerated mock type for the Client type
+type Client struct {
 	mock.Mock
 }
 
 // GetBlockByNumber provides a mock function with given fields: hex
-func (_m *EthClient) GetBlockByNumber(hex string) (models.BlockHeader, error) {
+func (_m *Client) GetBlockByNumber(hex string) (eth.BlockHeader, error) {
```
Comment (felder-cl) [inline core/internal/mocks/eth_client.go]: There may have been a mistake here with the renaming, as we're no longer in the package 'eth'

File: core/internal/mocks/tx_manager.go [hunk 3]
```diff
@@ -291,15 +301,15 @@ func (_m *TxManager) GetLINKBalance(address common.Address) (*assets.Link, error
 }
 
 // GetLogs provides a mock function with given fields: q
-func (_m *TxManager) GetLogs(q ethereum.FilterQuery) ([]models.Log, error) {
+func (_m *TxManager) GetLogs(q ethereum.FilterQuery) ([]eth.Log, error) {
 	ret := _m.Called(q)
 
-	var r0 []models.Log
-	if rf, ok := ret.Get(0).(func(ethereum.FilterQuery) []models.Log); ok {
+	var r0 []eth.Log
```
Comment (felder-cl) [inline core/internal/mocks/tx_manager.go]: This makes so much more sense now.

File: core/internal/mocks/eth_client.go (hunk 2 above)
Comment (dimroc) [reply core/internal/mocks/eth_client.go]: This is a special case because of `mocks`. All `mocks` of types found in other packages are placed here.

Comment (felder-cl) [reply core/internal/mocks/eth_client.go]: Oh I understand.  I mean should it still be called 'EthClient' here? because the package name isn't eth.

Comment (dimroc) [reply core/internal/mocks/eth_client.go]: Conceptually I see where you're coming from. But this is autogenerated code originating from `mockery`, and its convention is for the mock to have the same name as the interface. We use `go generate` to produce it, but even if I use the `mockery` CLI, I don't think there's an option to change the name of the output type:

```
Usage of mockery:
  -all
    	generates mocks for all found interfaces in all sub-directories
  -case string
    	name the mocked file using casing convention [camel, snake, underscore] (default "camel")
  -cpuprofile string
    	write cpu profile to file
  -dir string
    	directory to search for interfaces (default ".")
  -inpkg
    	generate a mock that goes inside the original package
  -keeptree
    	keep the tree structure of the original interface files into a different repository. Must be used with XX
  -name string
    	name or matching regular expression of interface to generate mock for
  -note string
    	comment to insert into prologue of each generated file
  -outpkg string
    	name of generated package (default "mocks")
  -output string
    	directory to write mocks to (default "./mocks")
  -print
    	print the generated mock to stdout
  -quiet
    	suppress output to stdout
  -recursive
    	recurse search into sub-directories
  -tags string
    	space-separated list of additional build tags to use
  -testonly
    	generate a mock in a _test.go file
  -version
    	prints the installed version of mockery
```

`-name` is for the origin interface, not the output mock.

**Pull request merged**